src/subtree-resolver/index.js

```diff
@@ -53,7 +54,7 @@ const resolveSubtree = async (obj, path, options = {}) => {
     ...resolveOptions,
     spec: normalized,
     allowMetaPatches: true,
-    skipNormalization: true,
+    skipNormalization: !isOpenAPI31(normalized),
```

Comment:
This shouldn't be needed. The OpenAPI 3.1.0 strategy should just ignore this setting as the strategy knows best what to do to achieve the goal.

`skipNormalization` is just going to be used for OpenAPI 3.0.x and OpenAPI 2.0, until they are replaced by new ApiDOM strategies.

Reply:
IMHO we need to address overall architecture in the light of the fact, that in order for have correctly normalized documents, normalization must always be performed after the dereference/resolution.

```js
const normalized = strategy.normalize(resolveOptions);
```

This doesn't make sense anymore.

Let me think some more to figure out how architecture needs to be changed to accommodate new realities.

Reply:
Both of these comments no longer apply.

We need to always perform normalization on original definition before we do lazy resolution (pre-normalization) . And then after lazy resolution perform the normalization again (post-normalization).
When lazy resolution is requested again on the same definition only post-normalization is performed.
It needs to work in this way to properly handle following case -if we perform lazy resolution without pre-normalization on a certain path e.g. `[paths, /, get]` - then the post-normalization can introduce an unresolved reference. (because it can copy e.g. parameters from unresolved/undereferenced part of the definition)